js_modules/dagster-ui/packages/ui-core/src/util/idb-lru-cache.ts

```diff
@@ -26,25 +27,26 @@ class IDBLRUCache<T> {
   private dbPromise: Promise<IDBDatabase> | undefined;
   private isDbOpen = false;
   private lruCache: LRUCache<T>;
+  private dbVersion?: number;
 
-  constructor({dbName, maxCount}: CacheOptions) {
-    this.dbName = dbName;
+  constructor({dbName, dbVersion, maxCount}: CacheOptions) {
+    this.dbName = `idb-lru-cache-v1-${dbName}`;
     this.maxCount = maxCount;
     this.lruCache = new LRUCache<T>(maxCount);
     this.dbPromise = this.initDB();
+    this.dbVersion = dbVersion;
   }
 
   private async initDB(): Promise<IDBDatabase> {
-    return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.dbName, 3);
+    const request = indexedDB.open(this.dbName, this.dbVersion);
```

Comment:
Moved this out of the Promise since this is the only behavioral change we've made.

Before the previous library used `openDB` from `idb` https://github.com/jakearchibald/idb/blob/main/src/entry.ts#L70 which has the request happen outside of the promise which I think will allow certain errors to be thrown synchronously trigger our try/catch behavior (around callsites of `cache`) which will trigger fallback indexeddb-less behavior